src/library/basesqltablemodel.cpp

```diff
@@ -842,7 +842,12 @@ void BaseSqlTableModel::setTrackValueForColumn(TrackPointer pTrack, int column,
     } else if (fieldIndex(ColumnCache::COLUMN_LIBRARYTABLE_TRACKNUMBER) == column) {
         pTrack->setTrackNumber(value.toString());
     } else if (fieldIndex(ColumnCache::COLUMN_LIBRARYTABLE_LOCATION) == column) {
-        pTrack->setLocation(value.toString());
+        DEBUG_ASSERT_AND_HANDLE(pTrack->getLocation() == value.toString()) {
+            qWarning() << "The location of track cannot be modified from"
+                    << pTrack->getLocation()
+                    << "to"
+                    << value.toString();
+        }
```

Comment:
IMHO we can be more strict here and assert if one tries to modify the location.
DEBUG_ASSERT_AND_HANDLE(false) 
This would be called if a user edits the location column. 
Why would a user do that?